src/config/storage/json.rs

```diff
@@ -1,25 +1,24 @@
 use crate::config::settings::Setting;
-use crate::config::Store;
+use crate::config::StorageAdapter;
 use crate::types::{Error, Result};
 use log::warn;
 use serde_json::Value;
 use std::collections::HashMap;
 use std::fs;
 use std::fs::File;
 use std::io::{Read, Seek, Write};
-use std::sync::{Arc, Mutex};
+use std::sync::Mutex;
 
 pub struct JsonStorageAdapter {
     path: String,
-    file_mutex: Arc<Mutex<File>>,
-    elements: HashMap<String, Setting>,
+    elements: Mutex<HashMap<String, Setting>>,
```

Comment:
Mutex to guard against multiple threads. This needs to be fixed better when saving the json